filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilderFactory.scala

```diff
@@ -37,7 +37,7 @@ final case class BlobPathBuilderFactory(globalConfig: Config, instanceConfig: Co
 }
 
 sealed trait BlobTokenGenerator {
-  def getAccessToken: String
+  def getAccessToken: AzureSasCredential
```

Comment:
This PR made me realize that we should rename this method. `get` makes it seem like a regular Java getter. Maybe something like `generateAccessToken`?

I also think we should change this method signature to return a `Try[AzureSasCredential]` rather than throwing from this method. 

Apologies for the somewhat out-of-scope feedback, but I think the return type change will affect the refresh logic.

Reply:
I made an attempt for this, but am struggling with a situation where I have nested try declarations that I am unsure how to flatten. It seems a little like one way or another this exception needs to feed into the top level try